### PR TITLE
Organizations: validate that a correct slug is generated

### PR DESCRIPTION
**Pull request by stsewd**
- Check for empty slugs
- Check for a max length of 32
  (enforced to the form level for now)

### CODE REVIEW

File: readthedocs/organizations/forms.py [hunk 1]
```diff
@@ -57,7 +62,11 @@ def clean_name(self):
         name = self.cleaned_data['name']
         if self.instance and self.instance.name and name == self.instance.name:
             return name
-        if Organization.objects.filter(slug=slugify(name)).exists():
+
+        potential_slug = slugify(name)
+        if not potential_slug:
+            raise forms.ValidationError(_('Invalid organization name'))
```
Comment (ericholscher) [inline readthedocs/organizations/forms.py:68]:
```suggestion
            raise forms.ValidationError(_('Invalid organization name: no slug generated'))
```

This at least gives the user an idea how to fix it, but using `slug` is not great :/

File: readthedocs/organizations/tests/test_forms.py [hunk 2]
```diff
@@ -120,6 +120,24 @@ def test_add_duplicate_invite_by_email(self):
 
 class OrganizationSignupTest(OrganizationTestCase):
 
+    def test_create_organization_with_empy_slug(self):
+        data = {
+            'name': '往事',
```
Comment (ericholscher) [inline readthedocs/organizations/tests/test_forms.py:125]: I feel like we should be smarter with slugging here -- can we not generate a slug for other non-ASCII chars? I thought we had a solution for this.

**Review by ericholscher — APPROVED** [review]
This is more valid, but we should find a way to support non-ASCII org names.

File: readthedocs/organizations/tests/test_forms.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/organizations/tests/test_forms.py:125]: Same, but looks like we already are using the slugify function we are using from projects, but that only applies to chars close to roman letters (like á -> a), for chinese and japanese there isn't a direct mapping, we could research if there is something like in python https://en.wikipedia.org/wiki/Romanization_of_Japanese

**Pull request merged**